First upload of the RP2040 display

th0mpy opened the pull request:
This is my remix of the V0_display, based on the RP2040 (RaspberryPi Pico).

th0mpy commented:
This is my remixed version of the V0_Display, it's been tested and built into a Voron 0.2 and working successfully.

timmit99 reviewed (approved):
Files validated last night on my desktop, @bartlammers's branch has already been made and tested, good to approve!

timmit99 commented:
Got an error on the CLI action:
![image](https://github.com/VoronDesign/Voron-Hardware/assets/8399317/2778eac4-9d46-40e1-ba76-fd204480b907)

timmit99 commented on V0_Display_RP2040/STLs/README.md:
```diff
@@ -0,0 +1,23 @@
+# V0 Display Enclosure #
+![V-Naught_Display_Enclosure](../Images/V0pt1_Enclosure.png )
```
Looks like this image needs to be updated, as this doesn't exist in the current release. Looks like you just copy/pasted my original V0 display enclosure instructions without updating/copying over the needed image.

timmit99 reviewed (commented):
After CLI run, one error noticed.

bartlammers commented:
Few points of feedback:

- You changed the USB from horizontal to vertical, but not the STL files to match. To prevent this issue, is it possible perhaps to make it compatible with the existing V0 display layout? If not also fine, but the printable files will have to match then.
- The rear has no ground pour.
- The buttons are changed from through-hole to SMD ones. The readme does not reflect that yet.
- The screenshot in the readme is of the regular V0 display, not your version.
- Text on the silkscreen is in Play, but not Play bold as is used in the Voron typography (minor issue, just noticed it)
- If it is going to be merged into the VoronHardware repository, we would prefer to not have other logos on there besides an optional Voron logo. You can of course add your (nick)name on there instead for credit on the design.

th0mpy commented:
Don't commit the changes just yet. Just a few housecleaning things to get out of the way first. In the meantime, I updated per @bartlammers suggestions. 
1. Updated (rolled back) to micro-usb for usability in both a vertical and horizontal configuration. Later update may go back to USB-C
2. One 3.3v and one GND pour, coming up. 
3. Did some small updates to the README but will finish up with the housecleaning later (need to update pictures and such)
4. See point 3 😄 
5. SHOULD* be all updated to PLAY BOLD. Question, do you want knockout or prefer the regular font? 
6. My logo is removed (hidden in another layer) and the Voron logo is added. 

I'll get to the screenshots tonight and will push a new update as well as do just a little cleaning to tidy up the structure of the KiCad and regenerate the production files.

th0mpy commented:
I made a couple updates to the schematics, per some design considerations. 

- Added a 10k pull up on the CS line to make boot-up more reliable. 
- Moved the 1k resistor for BOOTSEL closer to the chip, reducing crosstalk on the xtal and cs lines. 
- Added 3 gpio pins, consistent with the original V0 display, in the same location as the previous design.

th0mpy commented:
@bartlammers all updates accounted for. Please review and let me know what you think.

timmit99 commented on V0_Display/KiCad/V0_Display.kicad_pro:
I think this "new" file is a conversion to anew version of KiCad. This can be removed for now @th0mpy as its not related to your board!

th0mpy replied:
That's correct, it's using the standard naming for KiCad 7x. I'm pretty sure it can't be changed.

timmit99 commented on V0_Display_RP2040/Images/IMG_1685.jpeg:
Can you change this files names to a more descriptive name?

timmit99 commented on V0_Display_RP2040/Images/IMG_1683.jpeg:
Can you change this files names to a more descriptive name?

timmit99 commented on V0_Display_RP2040/Images/IMG_1687.jpeg:
Can you change this files names to a more descriptive name?

timmit99 commented on V0_Display_RP2040/Images/IMG_1688.jpeg:
Can you change this files names to a more descriptive name?

timmit99 commented on V0_Display_RP2040/Images/IMG_1691.jpeg:
Can you change this files names to a more descriptive name?

timmit99 reviewed (changes requested):
2 things:
1) Would it make sense to rename the KiCad files (.kicad_pcb, .kicad_sch, and .kicad_pro) from `V0_Display` to `V0_Display_RP2040`?
2) For the STL's, If you reverted to the horizontal USB then the STLs should be 95% compatible with the stock right? (Only difference would be the back for the RP2040 specific boot buttons) If so, we can remove all duplicate files to avoid confusion!

th0mpy commented:
Regarding the STLs. Yes, the Voron 0 STLs are 100% the same as before. I can just remove those STLs.

th0mpy commented:
I just received a few boards yesterday with all of the changes. I'll snap a few new pics tonight and upload with updated filenames.

timmit99 commented:
@th0mpy 
Only thing i noticed was the Neopixel out silkscreen looks to overlap. May be best to make the pin names a bit larger and shift the title up a bit.
![image](https://github.com/VoronDesign/Voron-Hardware/assets/8399317/54c30926-eb80-4798-827f-97c619c8597c)
(this may just be my font though, but your image looks to have the same issue)

![image](https://github.com/VoronDesign/Voron-Hardware/assets/8399317/a225d7fb-f8da-4268-8711-0f0765017223)

th0mpy commented:
I updated that label.

timmit99 commented:
> I updated that label.

ok great, than i see nothing else to nitpick =D

wlhlm commented:
This PR seems to have removed [`V0_Display/KiCad/V0_Display.kicad_pcb`](https://github.com/VoronDesign/Voron-Hardware/commit/7cb8fe357bf74e6c0283adbdc65192968ab00ef1#diff-d3a1cceef0a6d4d2cdde632418e89cd517c667bed5f247873d316e5688d69ce0), which was the PCB layout for the STM32 V0 display. Was this intentional?

th0mpy commented:
The V0_Display.kicad_pcb was an accident. I've fixed it, but it still shows as a deletion in the PR.

th0mpy commented:
Although, I do see that the file was indeed deleted. I can do a new pull request to fix that.
